### PR TITLE
Swap fixes

### PR DESCRIPTION
**Pull request by letier3110**
https://www.notion.so/madfissolutions/Swap-Bugs-35ce46db77954d5a8985077f43424b8f

### CODE REVIEW

File: src/screens/swap/swap.tsx [hunk 1]
```diff
@@ -70,38 +58,35 @@ export const SwapScreen: FC = () => {
       kind: OpKind.TRANSACTION
     }));
 
+    console.log('opParams', opParams);
```
Comment (unixvb) [inline src/screens/swap/swap.tsx:61]: console.log

File: src/screens/swap/swap-form.tsx [hunk 2]
```diff
@@ -79,6 +88,42 @@ export const SwapForm: FC = () => {
     slippageTolerance
   );
 
+  // const assetsListWithTez = useMemo<TokenInterface[]>(
+  //   () => [tezosToken, ...assetsList].filter(x => !isDefined(x.artifactUri)),
+  //   [tezosToken, assetsList]
+  // );
```
Comment (unixvb) [inline src/screens/swap/swap-form.tsx:94]: delete

File: src/screens/swap/swap-form.tsx [hunk 3]
```diff
@@ -79,6 +88,42 @@ export const SwapForm: FC = () => {
...
+      }
+
+      setAssetsListWithTez(result);
+    } else {
+      setAssetsListWithTez(sourceArray);
+    }
+  }, [tezosToken, searchValue, assetsList]);
+
+  // const { setSearchValue: setSearchTezAssetsValue } = useSearchAssets(assetsListWithTez, setAssetsListWithTez, []);
+
```
Comment (unixvb) [inline src/screens/swap/swap-form.tsx:126]: delete

File: src/components/asset-amount-input/asset-amount-input.props.ts [hunk 4]
```diff
@@ -11,6 +11,15 @@ export interface AssetAmountInputProps {
...
   isError?: boolean;
   editable?: boolean;
+  toUsdToggle?: boolean;
+  isSearchable?: boolean;
+  selectionOptions?:
+    | {
+        start: number;
+        end?: number | undefined;
+      }
+    | undefined;
```
Comment (unixvb) [inline src/components/asset-amount-input/asset-amount-input.props.ts:21]: extend this property from `TextInputProps`

File: src/form/form-asset-amount-input/form-asset-amount-input.tsx [hunk 5]
```diff
@@ -11,15 +11,28 @@ interface Props
...
   name: string;
+  toUsdToggle?: boolean;
+  isSearchable?: boolean;
+  setSearchValue?: EventFn<string>;
+  selectionOptions?:
+    | {
+        start: number;
+        end?: number | undefined;
+      }
+    | undefined;
```
Comment (unixvb) [inline src/form/form-asset-amount-input/form-asset-amount-input.tsx:22]: Extend `Pick<AssetAmountInputProps>` properties

File: src/screens/swap/swap-form.tsx [hunk 6]
```diff
@@ -79,6 +88,42 @@ export const SwapForm: FC = () => {
...
+        ) {
+          result.push(asset);
+        }
+      }
+
+      setAssetsListWithTez(result);
+    } else {
+      setAssetsListWithTez(sourceArray);
+    }
+  }, [tezosToken, searchValue, assetsList]);
```
Comment (unixvb) [inline src/screens/swap/swap-form.tsx:123]: maybe it's better to use `useMemo` here?

File: src/form/form-asset-amount-input/form-asset-amount-input.tsx [hunk 7]
```diff
@@ -9,17 +9,25 @@ import { ErrorMessage } from '../error-message/error-message';
 
 interface Props
   extends Omit<AssetAmountInputProps, 'value' | 'onValueChange'>,
-    Partial<Pick<AssetAmountInputProps, 'onValueChange'>> {
+    Partial<Pick<AssetAmountInputProps, 'onValueChange'>>,
+    Pick<AssetAmountInputProps, 'selectionOptions'> {
   name: string;
+  toUsdToggle?: boolean;
+  isSearchable?: boolean;
```
Comment (unixvb) [inline src/form/form-asset-amount-input/form-asset-amount-input.tsx:16]: this 2 fields also could be Picked from AssetAmountInputProps

**Review by unixvb — APPROVED** [review]

**Review by LendiHop — APPROVED** [review]

**Pull request merged**